js/src/blocks/product-select-field/block.json

```diff
@@ -0,0 +1,37 @@
+{
+	"$schema": "https://schemas.wp.org/trunk/block.json",
+	"apiVersion": 2,
+	"name": "google-listings-and-ads/product-select-field",
+	"title": "Product select control",
+	"textdomain": "google-listings-and-ads",
+	"attributes": {
+		"_templateBlockHideConditions": {
+			"type": "array"
+		},
```

Comment:
`__experimentalRole` is an undocumented attribute property, and is commonly set in WooCommerce core, but it seems a bit paradoxical. Although I did find [the use of its source code](https://github.com/WordPress/gutenberg/blob/125c130818ebb83c50618722858e8315d90e76b8/packages/blocks/src/store/selectors.js#L796-L815), it's not clear what it's used for, only knew that: without this property, it may lead to a custom block in editing-disabled mode.

As the product block editor in Woo core is still under development, this issue may be/has been resolved. However, it would be safer to keep it for the time being.

Reply:
Thanks for the explanation. I was referring to 

```
"_templateBlockHideConditions": {
			"type": "array"
		},
```

Reply:
Sorry, I got it wrong.

In order to make a custom block compatible with [the conditionally hiding feature](https://github.com/woocommerce/woocommerce/blob/8.4.0/plugins/woocommerce/src/Admin/Features/ProductBlockEditor/ProductTemplates/README.md#conditionally-hiding-a-block-in-product-editor-templates), it requires the `_templateBlockHideConditions` attribute when registering the custom block on both PHP and JS sides.

On JS side, a custom block can use [registerProductEditorBlockType](https://github.com/woocommerce/woocommerce/blob/8.4.0/packages/js/product-editor/src/utils/register-product-editor-block-type.ts#L58) to [inject that attribute](https://github.com/woocommerce/woocommerce/blob/8.4.0/packages/js/block-templates/src/utils/register-woo-block-type.ts#L92-L123), but on PHP side, there is no corresponding way to do so. Therefore, a custom still needs to do a similar injection as [BlockRegistry::register_block](https://github.com/woocommerce/woocommerce/blob/8.4.0/plugins/woocommerce/src/Admin/Features/ProductBlockEditor/BlockRegistry.php#L210-L213) when calling GB `register_block_type`.

The good news is Woo 8.5.0 will have the core support by [PR 41973 - Template API: Registration of custom block types](https://github.com/woocommerce/woocommerce/pull/41937), so the `_templateBlockHideConditions` attribute will be removed in a subsequent PR.